src/connectors/onchain_events/mod.rs

```diff
@@ -0,0 +1,361 @@
+use std::collections::HashMap;
+
+use alloy::{
+    primitives::{address, Address, Bytes, FixedBytes, Uint},
+    providers::{Provider, ProviderBuilder, RootProvider},
+    rpc::types::{Filter, Log},
+    sol,
+    sol_types::SolEvent,
+    transports::http::{Client, Http},
+};
+use futures_util::stream::StreamExt;
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+use tracing::{debug, error};
+
+sol!(
+    #[allow(missing_docs)]
+    #[sol(rpc)]
+    StorageRegistryAbi,
+    "src/connectors/onchain_events/storage_registry_abi.json"
+);
+
+sol!(
+    #[allow(missing_docs)]
+    #[sol(rpc)]
+    IdRegistryAbi,
+    "src/connectors/onchain_events/id_registry_abi.json"
+);
+
+sol!(
+    #[allow(missing_docs)]
+    #[sol(rpc)]
+    KeyRegistryAbi,
+    "src/connectors/onchain_events/key_registry_abi.json"
+);
+
+static STORAGE_REGISTRY: Address = address!("00000000fcce7f938e7ae6d3c335bd6a1a7c593d");
+
+static KEY_REGISTRY: Address = address!("00000000Fc1237824fb747aBDE0FF18990E59b7e");
+
+static ID_REGISTRY: Address = address!("00000000Fc6c5F01Fc30151999387Bb99A9f489b");
+
+static CHAIN_ID: u64 = 10; // OP mainnet
+const RENT_EXPIRY_IN_SECONDS: u64 = 365 * 24 * 60 * 60; // One year
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Config {
+    pub rpc_url: String,
+    pub disable: bool,
+}
+
+impl Default for Config {
+    fn default() -> Config {
+        return Config {
+            rpc_url: "must specify rpc url in config".to_string(),
```

Comment:
I noticed when this is set to a invalid url, the program exits with:

Error: UnableToParseUrl(RelativeUrlWithoutBase)

and a little hard to debug.

As a separate idea, maybe the default value can be empty (or make it optional) and then do a check at startup (but only when the component is enabled)